src/Tribe/Views/V2/Query/Query.php

```diff
@@ -0,0 +1,225 @@
+<?php
+/**
+ * Handles queries specific to Views v2.
+ *
+ * @since   TBD
+ *
+ * @package Tribe\Events\Views\V2\Query
+ */
+
+namespace Tribe\Events\Views\V2\Query;
+
+use Tribe__Cache_Listener as Cache_Listener;
+use Tribe__Date_Utils as Dates;
+use Tribe__Timezones as Timezones;
+use Tribe__Utils__Array as Arr;
+
+/**
+ * Class Query
+ *
+ * @since   TBD
+ *
+ * @package Tribe\Events\Views\V2\Query
+ */
+class Query {
+	/**
+	 * The prefix used to store the values cached by this class.
+	 *
+	 * @since TBD
+	 */
+	const CACHE_PREFIX = 'tribe_days_';
+
+	/**
+	 * Fetches the event-relevant post information of all events for a period and updates the `tribe_days` cache.
+	 *
+	 * The result of this method, the one stored in cache ,is a raw one, code using the results of this query is
+	 * supposed to manipulate the data to order, filter and sort it!
+	 * The method will cache the results for each day in the `tribe_days` cache.
+	 * The results stored in cache, each an array, for the period, not sorted, not ordered, are instances of the
+	 * `Event_Result` class.
+	 *
+	 * @since TBD
+	 *
+	 * @param \DateTimeInterface $start The start date of the period, correct time should be set already.
+	 * @param \DateTimeInterface $end The end date of the period, correct time should be set already.
+	 */
+	public static function update_period_cache( \DateTimeInterface $start, \DateTimeInterface $end ) {
+		global $wpdb;
+
```

Comment:
We should have a filter here to be able to bypass this whole method.

Reply:
This will be moved into the repository and that will be taken care of using the default extension system of the repository.
This would need to be filtered if it stayed like this, I agree.